test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/testproviders/HttpsWorkbenchTestProvider.java

```diff
@@ -145,6 +146,92 @@ public void testControllerOperations(WorkbenchDeployment workbenchDeployment, bo
+    public void testAdminUserPasswordChange(WorkbenchDeployment workbenchDeployment) {
+        final String NEW_PASSWORD = "newpassword1234!";
+        try {
+            // Check if old user can connect
+            HttpResponseParser responseCheckServerTemplate = getListServerTemplatesResponse(workbenchDeployment, workbenchDeployment.getUsername(), workbenchDeployment.getPassword());
+            assertThat(responseCheckServerTemplate.code()).isEqualTo(HttpsURLConnection.HTTP_OK);
+
+            logger.info("Change password to {} in credential secret", NEW_PASSWORD);
+            workbenchDeployment.changePassword(NEW_PASSWORD);
+
+            SoftAssertions.assertSoftly(softly -> {
+                try {
+                    // Check if user with old password cannot connect
+                    softly.assertThat(getListServerTemplatesResponse(workbenchDeployment, workbenchDeployment.getUsername(), workbenchDeployment.getPassword()).code())
+                          .as("Response code after login with old password")
+                          .isEqualTo(HttpsURLConnection.HTTP_UNAUTHORIZED);
+
+                    // Check if user with new password can connect  
+                    softly.assertThat(getListServerTemplatesResponse(workbenchDeployment, workbenchDeployment.getUsername(), NEW_PASSWORD).code())
+                          .as("Response code after login with new password")
+                          .isEqualTo(HttpsURLConnection.HTTP_OK);
+                } catch (IOException e) {
+                    logger.error("Unable to connect to workbench REST API when using SoftAssertions", e);
+                    throw new UncheckedIOException("Unable to connect to workbench REST API when using SoftAssertions", e);
+                }
+            });
+        } catch (IOException e) {
+            logger.error("Unable to connect to workbench REST API", e);
+            throw new RuntimeException("Unable to connect to workbench REST API", e);
+        } finally {
+            workbenchDeployment.changePassword(workbenchDeployment.getPassword());
+        }
+    }
+
+    // Verifies https://issues.redhat.com/browse/RHPAM-2777
+    public void testAdminUserNameAndPasswordChange(WorkbenchDeployment workbenchDeployment) {
+        final String NEW_USERNAME = "newadminusername";
+        final String NEW_PASSWORD = "newpassword4321!";
+        try {
+            // Check if old user can connect
+            HttpResponseParser responseCheckServerTemplate = getListServerTemplatesResponse(workbenchDeployment, workbenchDeployment.getUsername(), workbenchDeployment.getPassword());
+            assertThat(responseCheckServerTemplate.code()).isEqualTo(HttpsURLConnection.HTTP_OK);
+
+            logger.info("Change username to {} and password to {} in credential secret", NEW_USERNAME, NEW_PASSWORD);
+            workbenchDeployment.changeUsernameAndPassword(NEW_USERNAME, NEW_PASSWORD);
+
+            SoftAssertions.assertSoftly(softly -> {
```

Comment:
I think we could reduce the code duplication and make both tests more readable by doing something like:

{code}
private Consumer<SoftAssertions> isUserAuthorizedInWorkbench(String username, String password) {
...
}

private Consumer<SoftAssertions> isUserUnauthorizedInWorkbench(String username, String password) {
...
}
{code}

And then compose the asserts in this method and when only password by:

{code}
SoftAssertions.assertSoftly(isUserAuthorizedInWorkbench(username, NEW_PASSWORD).and(isUserUnauthorizedInWorkbench(username, NEW_PASSWORD))
{code}

What do you think?